test/e2e/specs/editor/various/unsynced-pattern.spec.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Unsynced pattern', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'create a new unsynced pattern via the block options menu', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'A useful paragraph to reuse' },
+		} );
+
+		// Create an unsynced pattern from the paragraph block.
+		await editor.showBlockToolbar();
+		await page
+			.getByRole( 'toolbar', { name: 'Block tools' } )
+			.getByRole( 'button', { name: 'Options' } )
+			.click();
+		await page.getByRole( 'menuitem', { name: 'Create pattern' } ).click();
+
+		const createPatternDialog = page.getByRole( 'dialog', {
+			name: 'Create pattern',
+		} );
+		await createPatternDialog
+			.getByRole( 'textbox', { name: 'Name' } )
+			.fill( 'My unsynced pattern' );
+		await page.getByLabel( 'Synced' ).click();
```

Comment:
It's recommended to use `getByRole` whenever possible so that it's clearer what type of element we're interacting with.

We can also use `setChecked` here so that we know which value it's set to. If we want to also test the default state then we can add a new assertion before clicking.

Reply:
Fixed - hadn't seen the `setChecked` option, handy to know.